primo/utils/override_utils.py

```diff
@@ -249,47 +168,190 @@ def assess_feasibility(self) -> bool:
 
         return True
 
-    def backfill(self) -> List[str]:
+
+# pylint: disable=too-many-instance-attributes
+class OverrideCampaign:
+    """
+    Class for constructing new campaigns based on the override results
+    and returning infeasibility information.
+
+    Parameters
+    ----------
+    override_selections : OverrideSelections
+        Object containing the override selections
+
+    opt_inputs : OptModelInputs
+        Object containing the necessary inputs for the optimization model
+
+    opt_campaign : dict
+        A dictionary for the original suggested P&A project
+        where keys are cluster numbers and values
+        are list of wells for each cluster.
+
+    eff_metrics : EfficiencyMetrics
+        The efficiency metrics
+    """
+
+    def __init__(
+        self,
+        override_selections,
+        opt_inputs,
+        opt_campaign: Dict,
+        eff_metrics,
+    ):
+        logging.getLogger("Campaign").setLevel(logging.WARNING)
+        opt_campaign_copy = copy.deepcopy(opt_campaign)
+        self.new_campaign = opt_campaign_copy
+        self.remove = override_selections.remove_widget_return
+        self.add = override_selections.add_widget_return
+        self.lock = override_selections.lock_widget_return
+        self.opt_inputs = opt_inputs
+        self.eff_metrics = eff_metrics
+
+        # change well cluster
+        self._modify_campaign()
+        self.plug_list = []
+        for _, well_list in self.new_campaign.items():
+            self.plug_list += well_list
+        self.wd = self.opt_inputs.config.well_data._construct_sub_data(self.plug_list)
+
+        self.feasibility = AssessFeasibility(
+            self.opt_inputs, self.new_campaign, self.wd, self.plug_list
+        )
+
+    def _modify_campaign(self):
+        """
+        Modify the original suggested P&A project
+        """
+        # remove clusters
+        for cluster in self.remove.cluster:
+            del self.new_campaign[cluster]
+
```

Comment:
also expecting a small change here to ensure that wells that are in projects that are deleted, but also selected to be locked will not throw any errors.

Reply:
I believe this won't cause any issues. This function is designed to update the existing suggested P&A projects, and any wells or clusters locked will not be considered during this step. The locked wells or clusters will only impact the re-optimization process.